modules/auth/password/hash/pbkdf2.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package hash
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"strings"
+
+	"code.gitea.io/gitea/modules/log"
+
+	"golang.org/x/crypto/pbkdf2"
+)
+
+func init() {
+	registerHasher("pbkdf2", NewPBKDF2Hasher)
+}
+
+// PBKDF2Hasher implements PasswordHasher
+// and uses the PBKDF2 key derivation function.
+type PBKDF2Hasher struct {
+	iter, keyLen int
+}
+
+// HashWithSaltBytes a provided password and salt
+func (hasher *PBKDF2Hasher) HashWithSaltBytes(password string, salt []byte) string {
+	if hasher == nil {
+		return ""
+	}
+	return hex.EncodeToString(pbkdf2.Key([]byte(password), salt, hasher.iter, hasher.keyLen, sha256.New))
+}
+
+// NewPBKDF2Hasher is a factory method to create an PBKDF2Hasher
+// config should be either empty or of the form:
+// "<iter>$<keyLen>", where <x> is the string representation
+// of an integer
+func NewPBKDF2Hasher(config string) *PBKDF2Hasher {
```

Comment:
```suggestion
// NewPBKDF2Hasher is a factory method to create an PBKDF2Hasher
// config should be either empty or of the form:
// "<iter>$<keyLen>", where <x> is the string representation
// of an integer
// Unlike the config default of pbkdf2 which is pbkdf2_v2(pbkdf2$320000$50),
// The default hasher of this function is pbkdf2_v1(pbkdf2$10000$50) to avoid breaking legacy passwords in database
func NewPBKDF2Hasher(config string) *PBKDF2Hasher {
```

It seems that the function  works like this?

Reply:
Yes but I don't think we need to write that comment there as it would be the same for every hasher I've added a number of comments elsewhere to hopefully make things a bit clearer. 

Including huge DO NOT CHANGE THIS warnings because you're right someone is going to come in and attempt to change things without understanding why things are written the way they are and why things are just a little more complex than it would at first appear.